### PR TITLE
upgrade tracy to 0.10.0

### PR DESCRIPTION
**Pull request by aimoonchen**
upgrade tracy to 0.10.0
E373D739-AC5C-450e-9B6B-B35A44C4CEDE

### CODE REVIEW

File: Source/Urho3D/Core/Profiler.h [hunk 1]
```diff
@@ -22,9 +22,9 @@
 
 #pragma once
 
-#include <tracy/Tracy.hpp>
+#include <tracy/public/tracy/Tracy.hpp>
```
Comment (rokups) [inline Source/Urho3D/Core/Profiler.h:25]: This is probably not necessary and can be fixed in CMakeLists.txt by providing a proper include path.

File: Source/ThirdParty/tracy/imgui/imgui.h
Comment (rokups) [inline Source/ThirdParty/tracy/imgui/imgui.h:1]: We are using our own imgui from ThirdParty. Please do not add a second copy here.

File: Source/ThirdParty/tracy/server/TracySourceView.cpp [hunk 2]
```diff
@@ -2,25 +2,23 @@
 #include <inttypes.h>
 #include <stdio.h>
 
-#include <capstone/capstone.h>  // rbfx
+#include <capstone.h>
```
Comment (rokups) [inline Source/ThirdParty/tracy/server/TracySourceView.cpp:5]: We have multiple patches marked with `// rbfx` and they must be preserved.

**Review by rokups — CHANGES_REQUESTED** [review]
Hi, thank you for taking an interest and submitting a PR. Things are never that simple, but i hope you can work out required changes :)

File: Source/Urho3D/Core/Profiler.h (hunk 1 above)
Comment (aimoonchen) [reply Source/Urho3D/Core/Profiler.h:25]: I just want to keep the directory structure style with upstream branch.

File: Source/ThirdParty/tracy/imgui/imgui.h
Comment (aimoonchen) [reply Source/ThirdParty/tracy/imgui/imgui.h:1]: Ok, i will try to remove unnecessary code.

File: Source/ThirdParty/tracy/server/TracySourceView.cpp (hunk 2 above)
Comment (aimoonchen) [reply Source/ThirdParty/tracy/server/TracySourceView.cpp:5]: Sorry, i didn't notice it.  i will check it more carefully later.

File: Source/Urho3D/Core/Profiler.h (hunk 1 above)
Comment (rokups) [reply Source/Urho3D/Core/Profiler.h:25]: You can keep directory structure. You need to modify `target_include_directories` in tracy `CMakeLists.txt` to include "tracy/public".

**Comment by rokups** [comment]
> 1. There is already an official application Tracy.exe, why do we need another TracyProfiler.exe app?

We build our own `Profiler.exe` in order to provide a full package with our releases and not complicate user experience. Besides, this way we know that our shipped version is fully compatible with our SDK. So indeed we need it.

> 2. Is it easy to upgrade imgui to latest version? because the latest tracy depend on some new features such as "InputEventsQueue":

Unfortunately this is going to be quite complicated. We have a set of non-trivial patches applied to handle HDPI screens. So seems the first step would be to rebase our fixes on the new imgui version, upgrade imgui and then upgrade tracy. Our custom fixes are on this branch: https://github.com/rbfx/imgui/commits/rbfx-thirdparty/

Profiler app gave me headaches in the past too. It used to work only with glfw backend and i had to heavily modify it to not pull in an extra dependency. I see it has been reworked considerably, and alternative and simpler approach would be implementing our own `BackendSDL.cpp`, basing it on existing `BackendGlfw.cpp`. It is acceptable to use SDL+OpenGL (default imgui opengl backend) on all desktop platforms for the profiler tool.

**Comment by aimoonchen** [comment]
Thanks for your reply, it seems that we need more considerations.

**Comment by aimoonchen** [comment]
No, what we have now works for me, i just thought upgrading tracy would be an easy task, thank you for your patience.

**Pull request closed without merging**